### PR TITLE
fix: pass WorkObject through to SubscribeNewHeads

### PR DESCRIPTION
**Pull request by alanorwick**
@dominant-strategies/core-dev

### CODE REVIEW

File: quai/filters/api.go [hunk 1]
```diff
@@ -266,14 +266,14 @@ func (api *PublicFilterAPI) NewHeads(ctx context.Context) (*rpc.Subscription, er
...
-		headers := make(chan *types.Header)
+		headers := make(chan *types.WorkObject)
 		headersSub := api.events.SubscribeNewHeads(headers)
 
 		for {
 			select {
 			case h := <-headers:
 				// Marshal the header data
-				marshalHeader := h.RPCMarshalHeader()
+				marshalHeader := h.Header().RPCMarshalHeader()
```
Comment (kiltsonfire) [inline quai/filters/api.go:276]: delete ".Header()"

File: quai/filters/filter_system.go [hunk 2]
```diff
@@ -361,7 +361,7 @@ func (es *EventSystem) handleTxsEvent(filters filterIndex, ev core.NewTxsEvent)
 
 func (es *EventSystem) handleChainEvent(filters filterIndex, ev core.ChainEvent) {
 	for _, f := range filters[BlocksSubscription] {
-		f.headers <- ev.Block.Header()
```
Comment (kiltsonfire) [inline quai/filters/filter_system.go:364]: delete .Header()

File: quai/filters/api.go (hunk 1 above)
Comment (alanorwick) [reply quai/filters/api.go]: fixed

File: quai/filters/filter_system.go (hunk 2 above)
Comment (alanorwick) [reply quai/filters/filter_system.go:364]: fixed

**Comment by alanorwick** [comment]
https://github.com/kiltsonfire/go-quai/pull/2/files

**Pull request closed without merging**